logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -123,8 +128,16 @@ public String toString() {
     private volatile Optional<Timestamp> oldestSegmentTimestamp;
     private volatile Optional<Path> oldestSegmentPath = Optional.empty();
     private final TemporalAmount retentionTime;
-
     private final SchedulerService flusherService;
+    private ScheduledExecutorService scheduledFSWatcher = Executors.newScheduledThreadPool(1, r -> {
+        Thread t = new Thread(r);
+        //Allow this thread to die when the JVM dies
+        t.setDaemon(true);
+        //Set the name
+        t.setName("dlq-fs-watcher");
```

Comment:
suggest to make the name `dlq-fs-watcher` more specific. We have watchers on read and write side, for  maintaining policy and notification from segment deletion. Maybe something like `dlq-segment-check`?